src/openrct2/object/RideObject.cpp

```diff
@@ -532,3 +524,642 @@ uint8 RideObject::CalculateNumHorizontalFrames(const rct_ride_entry_vehicle * ve
+            cars.push_back(car);
+        }
+    }
+    else if (json_is_object(jCars))
+    {
+        auto car = ReadJsonCar(jCars);
+        cars.push_back(car);
+    }
+    return cars;
+}
+
+rct_ride_entry_vehicle RideObject::ReadJsonCar(const json_t * jCar)
+{
+    rct_ride_entry_vehicle car = { 0 };
+    car.rotation_frame_mask = ObjectJsonHelpers::GetInteger(jCar, "rotationFrameMask");
+    car.spacing = ObjectJsonHelpers::GetInteger(jCar, "spacing");
+    car.car_mass = ObjectJsonHelpers::GetInteger(jCar, "mass");
+    car.tab_height = ObjectJsonHelpers::GetInteger(jCar, "tabOffset");
+    car.num_seats = ObjectJsonHelpers::GetInteger(jCar, "numSeats");
+    if (ObjectJsonHelpers::GetBoolean(jCar, "seatsInPairs"))
+    {
+        car.num_seats |= 0x80;
+    }
+
+    car.sprite_width = ObjectJsonHelpers::GetInteger(jCar, "spriteWidth");
+    car.sprite_height_negative = ObjectJsonHelpers::GetInteger(jCar, "spriteHeightNegative");
+    car.sprite_height_positive = ObjectJsonHelpers::GetInteger(jCar, "spriteHeightPositive");
+    car.animation = ObjectJsonHelpers::GetInteger(jCar, "animation");
+    car.base_num_frames = ObjectJsonHelpers::GetInteger(jCar, "baseNumFrames");
+    car.no_vehicle_images = ObjectJsonHelpers::GetInteger(jCar, "numImages");
+    car.no_seating_rows = ObjectJsonHelpers::GetInteger(jCar, "numSeatRows");
+    car.spinning_inertia = ObjectJsonHelpers::GetInteger(jCar, "spinningInertia");
+    car.spinning_friction = ObjectJsonHelpers::GetInteger(jCar, "spinningFriction");
+    car.friction_sound_id = ObjectJsonHelpers::GetInteger(jCar, "frictionSoundId", 255);
+    car.log_flume_reverser_vehicle_type = ObjectJsonHelpers::GetInteger(jCar, "logFlumeReverserVehicleType");
+    car.sound_range = ObjectJsonHelpers::GetInteger(jCar, "soundRange", 255);
+    car.double_sound_frequency = ObjectJsonHelpers::GetInteger(jCar, "doubleSoundFrequency");
+    car.powered_acceleration = ObjectJsonHelpers::GetInteger(jCar, "poweredAcceleration");
+    car.powered_max_speed = ObjectJsonHelpers::GetInteger(jCar, "poweredMaxSpeed");
+    car.car_visual = ObjectJsonHelpers::GetInteger(jCar, "carVisual");
+    car.effect_visual = ObjectJsonHelpers::GetInteger(jCar, "effectVisual", 1);
+    car.draw_order = ObjectJsonHelpers::GetInteger(jCar, "drawOrder");
+    car.num_vertical_frames_override = ObjectJsonHelpers::GetInteger(jCar, "numVerticalFramesOverride");
+
+    auto& peepLoadingPositions = car.peep_loading_positions;
+    auto jLoadingPositions = json_object_get(jCar, "loadingPositions");
+    if (json_is_array(jLoadingPositions))
+    {
+        auto arr = ObjectJsonHelpers::GetJsonIntegerArray(jLoadingPositions);
+        for (auto x : arr)
+        {
+            peepLoadingPositions.push_back(x);
+        }
+    }
+    else
+    {
+        auto jLoadingWaypoints = json_object_get(jCar, "loadingWaypoints");
+        if (json_is_array(jLoadingWaypoints))
+        {
+            car.flags |= VEHICLE_ENTRY_FLAG_26;
```

Comment:
Unnecessary/invalid? Gets overwritten in line 847: https://github.com/OpenRCT2/OpenRCT2/pull/7310/files#diff-05500f2b0f3c02e0b8d5ac0587228314R847